src/storage/txn/scheduler.rs

```diff
@@ -251,6 +269,40 @@ pub struct Scheduler {
     running_write_count: usize,
 }
 
+// Make clippy happy.
+type MultipleReturnValue = (Option<MvccLock>, Vec<(u64, Write)>, Vec<(u64, bool, Value)>);
+
+fn find_mvcc_infos_by_key(reader: &mut MvccReader,
+                          key: &Key,
+                          mut ts: u64)
+                          -> Result<MultipleReturnValue> {
+    let mut writes = vec![];
+    let mut values = vec![];
+    let lock = reader.load_lock(key)?;
+    loop {
+        let opt = reader.seek_write(key, ts)?;
+        match opt {
```

Comment:
I think you can process `None` first and return, this may reduce conditions in each `{}` and make it more clear to read.